platform/darwin/src/MGLStyle.mm

```diff
@@ -254,5 +257,21 @@ - (void)removeStyleClass:(NSString *)styleClass
     }
 }
 
+- (void)setImage:(MGLImage *)image forName:(NSString *)name
+{
+    if (image && name)
```

Comment:
The first form will crash unless `name` is `[NSNull null]` specifically. Since `name` is of type `NSString *` rather than `id`, there is only one way to avoid the crash: set `name` to `nil`. Calling any method on `nil` does nothing and resolves to a `NO` or 0.

The second form will crash unless `name` is `NULL`, which is the opposite of what you’d do in this case. Moreover, in Objective-C, `nil` is conventionally used for Objective-C objects, `NULL` is used for C pointers and double-pointers, and `Nil` is used for Objective-C `Class`es. I recommend reading [this NSHipster article](http://nshipster.com/nil/) to learn about the various ways to say “nothing” in Objective-C.

In this case, you can simply `NSAssert(name)`.